projects/debouncer/src/debouncer.c

```diff
@@ -0,0 +1,58 @@
+#include "debouncer.h"
+
+#include "log.h"
+#include "soft_timer.h"
+#include "stm32f0xx_interrupt.h"
+
+#define INTERRUPT_MASKING_DURATION_MS 50
+
+// This is the callback for the soft timer. Gets the state and compares with old state.
+// If states are equal, it means that we had a button input. So it runs the given callback.
+static void prv_debouncer_timer_callback(SoftTimerID timer_id, void *context) {
+  DebouncerInfo *debouncer_info = context;
+
+  GPIOState current_state;
+  gpio_get_state(&debouncer_info->address, &current_state);
+
+  if (current_state == debouncer_info->state) {
+    debouncer_info->callback(&debouncer_info->address, debouncer_info->context);
+  }
+
+  // stm32f0xx_interrupt_exti_mask_set(debouncer_info->address.pin, false);
```

Comment:
Please remove commented code - it doesn't serve a purpose anymore and clutters the file.